octoprint_backupscheduler/__init__.py

```diff
@@ -290,9 +288,8 @@ def _sendEmailNotification(self, subject, body):
             server._host = self._settings.get(["send_email", "smtp_server"])
             server.connect(self._settings.get(["send_email", "smtp_server"]), self._settings.get_int(["send_email", "smtp_port"]))
             server.ehlo()
-            if self._settings.get(["send_email", "smtp_user"]) != "" and os.environ.get("BACKUPSCHEDULER_SMTP_PASSWORD", False):
-                f = Fernet(base64.urlsafe_b64encode(to_bytes(self._settings.global_get(["server", "secretKey"]))))
-                server.login(self._settings.get(["send_email", "smtp_user"]), to_str(f.decrypt(os.environ.get("BACKUPSCHEDULER_SMTP_PASSWORD"))))
+            if self._settings.get(["send_email", "smtp_user"]) != "" and self._settings.get(["send_email", "smtp_password"]) != "":
```

Comment:
I deleted the last os.environ usage on the SMTP section but now the loaded SMTP password is not available even it is loaded in def on_settings_load(self)

Reply:
oh, good catch. I wonder if setting a variable to the function being called and then using that variable in the login command would work. 

to test I would set a breakpoint there and see if the variable is decrypting properly or not.

Reply:
maybe creating another function call to grab the password live from the file system and returning the decrypted password as string. then call that in both places for smtp login and on_settings_load.

Reply:
Taken from the description in the OctoPrint help it should work and the settings page shows the correct decrypted value. Perhaps it is a bug in OctoPrints setting interface. Using an older version 1.9.4 for development...started debugging OctoPrints interface bugs it is a little complex with the generic implementation.

Reply:
I think it might be a potential bug. That self._settings.get for smtp_password is just returning None. I've switched over to what I was mentioning as possible solution. Try those changes.

Reply:
I just tested with an app password in gmail and the email notifications are definitely working for me in my windows dev environment, which is running OctoPrint 1.10.2.

Reply:
Will open a bug for it also for the discussion whats are the thoughts behind. I checked the backup file and the plguins data folders are included in it so also the data.txt. Perhaps it would be less complex if using getter/setter for de-/encryption and store the encrypted password in the config.yaml? The secret used for encryption is not stored in the backup? 

Will test it with Gmail. From setting up a central SMTP forwarding I know that Gmail has some specialties in the handling.

Reply:
Ah, found the additional_excludes_hook. But anyhow, if it is correctly encrypted perhaps its an option to also restore the password in the explicit host.